File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/ExcelProcessor.cpp
#include <vector>
#include <string>
#include <cctype>
#include <algorithm>
#include <xlsx/Document.h>
#include <xlsx/Sheet.h>
#include <xlsx/Cell.h>

class ExcelProcessor {
public:
    ExcelProcessor() {}

    std::vector<std::vector<xlsx::Cell>> read_excel(const std::string& file_name) {
        try {
            xlsx::Document doc;
            auto& sheet = doc.worksheets().begin()->second; // Get the first worksheet
            std::vector<std::vector<xlsx::Cell>> data;
            for (const auto& row : sheet) {
                std::vector<xlsx::Cell> row_data;
                for (const auto& cell : row) {
                    row_data.push_back(cell);
                }
                data.push_back(row_data);
            }
            return data;
        } catch (...) {
            return std::vector<std::vector<xlsx::Cell>>();
        }
    }

    int write_excel(const std::vector<std::vector<xlsx::Cell>>& data, const std::string& file_name) {
        try {
            xlsx::Document doc;
            auto& sheet = doc.add_worksheet("Sheet1");
            for (int i = 0; i < data.size(); i++) {
                for (int j = 0; j < data[i].size(); j++) {
                    sheet->write(i, j, data[i][j]);
                }
            }
            doc.save(file_name);
            return 1;
        } catch (...) {
            return 0;
        }
    }

    std::pair<int, std::string> process_excel_data(int N, const std::string& save_file_name) {
        std::vector<std::vector<xlsx::Cell>> data = read_excel(save_file_name);
        if (data.empty() || N >= static_cast<int>(data[0].size())) {
            return {0, ""};
        }

        std::vector<std::vector<xlsx::Cell>> new_data;
        for (const auto& row : data) {
            std::vector<xlsx::Cell> new_row;
            for (int i = 0; i < row.size(); i++) {
                if (i == N) {
                    std::string cell_str = row[i].to_string();
                    bool is_digit = true;
                    if (!cell_str.empty()) {
                        for (char c : cell_str) {
                            if (!std::isdigit(static_cast<unsigned char>(c))) {
                                is_digit = false;
                                break;
                            }
                        }
                    }
                    if (is_digit) {
                        new_row.push_back(row[i]);
                    } else {
                        std::transform(cell_str.begin(), cell_str.end(), cell_str.begin(), ::toupper);
                        new_row.push_back(xlsx::Cell(cell_str));
                    }
                } else {
                    new_row.push_back(row[i]);
                }
            }
            new_data.push_back(new_row);
        }

        std::size_t pos = save_file_name.find_last_of('.');
        std::string new_file_name = save_file_name.substr(0, pos + 1) + "_process.xlsx";
        int success = write_excel(new_data, new_file_name);
        return {success, new_file_name};
    }
};

File: data/derived_cot/rq1_traces/translation/qwen/cpp_to_py/code_output/ExcelProcessor.py
import openpyxl
import os

class ExcelProcessor:
    def read_excel(self, file_name):
        try:
            wb = openpyxl.load_workbook(file_name)
            sheet = wb.active
            data = []
            for row in sheet.iter_rows(min_row=1, max_row=sheet.max_row, min_col=1, max_col=sheet.max_column, values_only=True):
                row_data = []
                for cell_value in row:
                    if cell_value is None:
                        continue
                    if isinstance(cell_value, str):
                        if cell_value.lstrip('-').isdigit() and cell_value != '-' and cell_value.lstrip('-').strip().isdigit():
                            try:
                                row_data.append(int(cell_value))
                            except ValueError:
                                row_data.append(cell_value)
                        else:
                            row_data.append(cell_value)
                    else:
                        row_data.append(cell_value)
                data.append(row_data)
            return data
        except Exception as e:
            return []

    def write_excel(self, data, file_name):
        try:
            wb = openpyxl.Workbook()
            sheet = wb.active
            sheet.title = "Sheet1"
            for row_idx, row in enumerate(data):
                for col_idx, cell_value in enumerate(row):
                    cell = sheet.cell(row=row_idx+1, column=col_idx+1)
                    if isinstance(cell_value, int):
                        cell.value = cell_value
                    elif isinstance(cell_value, str):
                        cell.value = cell_value
            wb.save(file_name)
            return 1
        except Exception as e:
            return 0

    def process_excel_data(self, N, save_file_name):
        data = self.read_excel(save_file_name)
        if not data or not data[0] or N >= len(data[0]):
            return (0, "")
        
        new_data = []
        for row in data:
            new_row = list(row)
            col_value = row[N]
            if isinstance(col_value, str) and col_value:
                if col_value[0].isdigit():
                    new_row.append(col_value[0])
                else:
                    s = col_value
                    for i in range(len(s)):
                        if s[i] > 'Z':
                            s_list = list(s)
                            if ord(s[i]) > 90:
                                s_list[i] = chr(ord(s[i]) - 32)
                            s = ''.join(s_list)
                    new_row.append(s)
            else:
                new_row.append(col_value)
            new_data.append(new_row)
        
        base = save_file_name[:save_file_name.rfind('.')] if '.' in save_file_name else save_file_name
        new_file_name = base + "_process.xlsx"
        success = self.write_excel(new_data, new_file_name)
        return (success, new_file_name)